docs/concepts.md

```diff
@@ -67,6 +68,96 @@ that are specific to a particular Node (like per-process UNIX file descriptors),
 so handle value 3 for Node A identify a different channel than handle value 3
 for Node B (unless there happens to be a coincidence of numbering).
 
+## Labels
+
+Labels are used to enforce information flow control (IFC) between nodes and
+channels in an Oak Application and the outside world.
+
+Each node and each channel is associated with a label at creation time. The
+label does not change any more after that (apart from explicit declassification
+operations, which are not currently supported).
+
+A label has two components: **secrecy** and **integrity**.
+
+Each component is represented as an (unordered) set of tags, and each tag refers
+to individual security principal in the system.
+
+The following types of security principals are supported in Oak:
+
+- **user**: specified by a bearer token, public key, or some other assertion of
+  a user identity
+- **computation**: specified by a measurement (hash) of the WebAssembly module
+  running within an Oak Node
+
+The least privileged label is "public_trusted" (also referred to as `⊥`,
+pronounced "bottom"), which corresponds to a node or channel which has only
+observed public data, and its inputs are not endorsed with any level of
+integrity; in this label, both secrecy and integrity components are empty sets.
+
+Given two labels `L_a` and `L_b`, `L_a ⊑ L_b` (pronounced "flows to") iff `L_a`
+is at least as strict as `L_b`, i.e. `b` has "more secrecy" and "less integrity"
+than `a`.
+
+Intuitively, data can only flow to nodes or channels with:
```

Comment:
Another way of saying the same thing as this that might weird people out less is:

> Intuitively, data can only flow from a source to a destination if:
- The destination is **at least as secret** as the source
- The source is **at least as trusted** as the destination

Reply:
Nice! I was indeed struggling to word such a simple statement, and your version is exactly what I was looking for! In particular, I could not think of the adjective corresponding to "integrity", which, as you suggest, is actually "trusted" :) Thanks a lot!